### PR TITLE
Feature/batched nvt nose hoover

### PR DESCRIPTION
**Pull request by thomasloux**
## Summary

Add support for batch systems for NVT Nosé Hoover #262 
Add tests: batched systems, identical results for batched systems and independent single run (with same initial state, most importantly the momenta).

I've also carefully checked that the behaviour is the same as the previous non batched implementation.

## Checklist

Before a pull request can be merged, the following items must be checked:

* [ ] Doc strings have been added in the [Google docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google).
* [X] Run [ruff](https://beta.ruff.rs/docs/rules/#pydocstyle-d) on your code.
* [X] Tests have been added for any new functionality or bug fixes.

We highly recommended installing the pre-commit hooks running in CI locally to speedup the development process. Simply run `pip install pre-commit && pre-commit install` to install the hooks which will check your code before each commit.

### CODE REVIEW

**Comment by thomasloux** [comment]
@orionarcher You can review, normally there are not big changes ! Let me know if you want me to modify some parts

**Comment by abhijeetgangan** [comment]
@thomasloux Thanks for the PR. I will take a look at it once we merge the API changes.

File: torch_sim/integrators/nvt.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import torch
 
-from torch_sim.integrators.md import (
+from torch_sim.integrators.md_batched import (
```
Comment (abhijeetgangan) [inline torch_sim/integrators/nvt.py]: Wrong import?

Comment (thomasloux) [reply torch_sim/integrators/nvt.py]: you are right, typo from development when I'm wanted to check the consistency of batched vs unbatched. Just sent a fix

File: torch_sim/integrators/nvt.py [hunk 2]
```diff
@@ -377,14 +377,14 @@ def nvt_nose_hoover_init(
...
-        n_atoms_per_system = torch.bincount(state.system_idx)
+        n_atoms_per_system = state.n_atoms_per_system
         dof_per_system = (
             n_atoms_per_system * state.positions.shape[-1]
         )  # n_atoms * n_dimensions
 
         # For now, sum the per-system DOF as chain expects a single int
         # This is a limitation that should be addressed in the chain implementation
-        total_dof = int(dof_per_system.sum().item())
+        # total_dof = int(dof_per_system.sum().item())
```
Comment (orionarcher) [inline torch_sim/integrators/nvt.py]: is this limitation still true?

File: torch_sim/integrators/nvt.py [hunk 3]
```diff
@@ -430,10 +430,10 @@ def nvt_nose_hoover_update(
         chain = state.chain
 
         # Update chain masses based on target temperature
-        chain = chain_fns.update_mass(chain, kT)
+        # chain = chain_fns.update_mass(chain, kT)
```
Comment (orionarcher) [inline torch_sim/integrators/nvt.py]: Maybe remove commented code before merge

File: torch_sim/models/orb.py [hunk 4]
```diff
@@ -416,7 +416,7 @@ def forward(self, state: ts.SimState | StateDict) -> dict[str, torch.Tensor]:
             if not model_has_direct_heads and prop == "stress":
                 continue
             _property = "energy" if prop == "free_energy" else prop
-            results[prop] = predictions[_property].squeeze()
+            results[prop] = predictions[_property]
```
Comment (orionarcher) [inline torch_sim/models/orb.py:425]: Since I believe this is fixing #247 could you mention the issue in this PR?

Comment (orionarcher) [reply torch_sim/models/orb.py:425]: nvm I just merged #247 instead.

Comment (orionarcher) [inline torch_sim/models/orb.py]: It'd be helpful to understand why the current tests aren't capturing this issue. If you could write a test that fails without this change that'd be really helpful.

File: torch_sim/integrators/nvt.py (hunk 3 above)
Comment (thomasloux) [reply torch_sim/integrators/nvt.py]: this shouldn't have been commented to be honest

File: torch_sim/models/orb.py (hunk 4 above)
Comment (thomasloux) [reply torch_sim/models/orb.py]: hey, I added a new model output shape test with only 1 system. There was a test for a batch input only.

**Comment by thomasloux** [comment]
By the way, I probably broke the npt nosé hoover. So I need to adapt in the same way the integrator.

File: tests/models/conftest.py [hunk 5]
```diff
@@ -228,3 +228,85 @@ def test_model_output_validation(
     # Rename the function to include the test name
     test_model_output_validation.__name__ = f"test_{model_fixture_name}_output_validation"
     return test_model_output_validation
+
+
+def make_validate_single_system_model_outputs_test(
```
Comment (CompRhys) [inline tests/models/conftest.py]: can this single case just be added to the main model interface test factory and then can avoid changes to all the model files?

Comment (thomasloux) [reply tests/models/conftest.py]: you're right, i've updated it!
it's way better like that.

**Comment by thomasloux** [comment]
Hey @abhijeetgangan, I've merged the API change. Tests are passing. Note: we may want to add a test for we hardcode the value of a simulation with all integrators that we have to check that we don't change the results among versions.

**Comment by abhijeetgangan** [comment]
Thanks a lot @thomasloux . I will go through it this weekend. For the tests; I agree that we need another check other than the stability checks. We could have numerical values of MD averages. I would also like to have integration tests comparing the values with LAMMPS. Could be a script that runs in CI with bunch of assertions.

**Comment by thomasloux** [comment]
Added a correction for #266. Need a careful review for that to make sure that I did not forget something. Decided to add a condition rather then expanding xi, p_xi and Q in a similar way than LAMMPS

**Comment by thomasloux** [comment]
This is ready to merge

File: tests/test_integrators.py [hunk 6]
```diff
@@ -295,6 +295,399 @@ def test_nvt_langevin_multi_kt(
...
+        energies.append(state.energy)
+        temperatures.append(temp / MetalUnits.temperature)
+        invariants.append(ts.nvt_nose_hoover_invariant(state, kT))
+
+    # Convert temperatures list to tensor
+    temperatures_tensor = torch.stack(temperatures)
+    temperatures_list = [t.tolist() for t in temperatures_tensor.T]
+    assert torch.allclose(
+        temperatures_tensor[-1],
+        torch.tensor([299.9910, 299.6800], dtype=dtype),
```
Comment (abhijeetgangan) [inline tests/test_integrators.py:332]: These specific temperature values correspond to some seed?

**Comment by abhijeetgangan** [comment]
~~Just one comment about the specific temperature values in NPT and NVT tests. Otherwise good to merge.~~ This is to keep expected behavior over versions and changes to integrators.
